### DISCUSSION TITLE
What open x-platform language + UI framework +IDE w/ designer and no scrap in 2023 ?

### DISCUSSION THREAD

**Discussion by TehAppKiller** [microsoft/microsoft-ui-xaml#8809] [General]
Hello,
Just a quick feedback from someone who jumped quickly from C/C++ WinForm to C# WPF and now trying WinUI 3.

¤ If WinForm was prehistory, especially compared to others framework, at least it was working quickly and easily

¤ C#/WPF was a quick progress in modern design ; but ClickOnce generated a lot of bugs (is it installed or not? is framework good? random error message? can it be upgraded? what's the installed version? really?, usw)

¤ C#/MahApps implicates suicide (working against the IDE)

¤ WinUI-3 is a all-in against developer:
- no designer (at least for the layout, guys...) -> you can all merge on ((Visual))Studio Code and avoid people spend money
- directly linked to MStore scrap and MSIX by default
- requires developer mode enable or does not work -> had to dig in forums to avoid that, no official documentation for unpackaged/standalone given
- not working

So, I tried to make a simple UI to display data in an Array for a little COM/USB sensor.
Install went fine even if we don't understand all what we install.
Create project, quite okay (has to know diff between '.net' and '.net platform' though)
Surprise no UI, even if 'Open w/ XAML designer' still exists.
Have to remove all packaged and msix stuff, some even manually + add manually no dependency in project
Try to add a DataGrid, does not exist in basic components...
Try to follow CommunityToolKit guide: need to add a nugget, manual References, and... does not work !

'+ WinUI 3 Gallery does not give a full page of capabilities (like any other UI framework), just a list of example

So,
What language, UI framework who are cross-platform and open to use in 2023 please ?
Cause we're at archaic time + lot of headaches
I thought professional people were working to make human-centered things instead of g33ky/garage/not-working ones...

Thanks (?)

**Comment by christosk92** [comment]
Avalonia
